{"questionTitle": "issue in Django 5.0.2 and django_rest_framework", "question": "<p>hi guys i have issue with reading sql data from postgresql with django.\ni created my model with manage.py inspectdb\ncommand from existing database.</p>\n<p>my model:</p>\n<pre><code>from django.db import models\n\nclass Regions(models.Model):\n    code = models.CharField(unique=True, max_length=4)\n    capital = models.CharField(max_length=10)\n    name = models.TextField(unique=True)\n\n    class Meta:\n        managed = False\n        db_table = &quot;regions&quot;\n</code></pre>\n<p>My serializer:</p>\n<pre><code>from rest_framework import serializers\n\n\nclass RegionsSerializer(serializers.ModelSerializer):\n    class Meta:\n        model = &quot;france.models.Regions&quot;\n        fields = &quot;__all__&quot;\n\n\n</code></pre>\n<p>my viewset:</p>\n<pre><code>from rest_framework import viewsets\nfrom rest_framework.response import Response\nfrom .models import Regions\nfrom .serializers import RegionsSerializer\n\n\nclass RegionsView(viewsets.ViewSet):\n    def list(self, request):\n        regions = Regions.objects.all()\n        serializer = RegionsSerializer(regions, many=True)\n        return Response(serializer.data)\n</code></pre>\n<p>and error i get:</p>\n<pre><code>AttributeError at /regions/\n'str' object has no attribute '_meta'\nRequest Method: GET\nRequest URL:    http://127.0.0.1:1000/regions/\nDjango Version: 5.0.4\nException Type: AttributeError\nException Value:    \n'str' object has no attribute '_meta'\nException Location: C:\\Projects\\play_ground\\python\\django_one\\venv\\Lib\\site-packages\\rest_framework\\utils\\model_meta.py, line 35, in get_field_info\nRaised during:  france.views.RegionsView\nPython Executable:  C:\\Projects\\play_ground\\python\\django_one\\venv\\Scripts\\python.exe\nPython Version: 3.12.2\nPython Path:    \n['C:\\\\Projects\\\\play_ground\\\\python\\\\django_one',\n 'C:\\\\Users\\\\safkh\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python312\\\\python312.zip',\n 'C:\\\\Users\\\\safkh\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python312\\\\DLLs',\n 'C:\\\\Users\\\\safkh\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python312\\\\Lib',\n 'C:\\\\Users\\\\safkh\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python312',\n 'C:\\\\Projects\\\\play_ground\\\\python\\\\django_one\\\\venv',\n 'C:\\\\Projects\\\\play_ground\\\\python\\\\django_one\\\\venv\\\\Lib\\\\site-packages']\nServer time:    Fri, 05 Apr 2024 08:19:46 +0000\n</code></pre>\n<p>can you help me with issue thnx all</p>\n", "acceptedAnswer": "<p>The serializer <code>Meta</code> can not work with a string as model, but should refer to the model class, so:</p>\n<pre><code>from france.models <b>import Regions</b>\nfrom rest_framework import serializers\n\n\nclass RegionsSerializer(serializers.ModelSerializer):\n    class Meta:\n        <b>model = Regions</b>\n        fields = '__all__'</code></pre>\n<hr />\n<blockquote>\n<p><strong>Note</strong>: normally a Django model is given a <em>singular</em> name, so <code>Region</code> instead of <s><code>Regions</code></s>.</p>\n</blockquote>\n", "questionSource": "https://stackoverflow.com/questions/78278665/issue-in-django-5-0-2-and-django-rest-framework"}